### PR TITLE
Add KServe and Triton inference adapters

### LINKED ISSUE #829: Add a model-repository adapter for KServe and Triton inference-server backends
## Summary
Organizations standardized on KServe/Triton want OpenMed token-classification families served from their existing inference fabric rather than the bundled runtime. A repository adapter that lays out exported models in the Triton model-repository layout lets the pipeline call out to a Triton/KServe endpoint as a pluggable backend.

## Scope
- [ ] Add an adapter that writes config.pbtxt and the model-repository directory layout for an exported ONNX token-classification family
- [ ] Add a remote inference backend that calls a Triton/KServe HTTP/gRPC endpoint and reshapes outputs into pipeline logits
- [ ] Make the backend selectable via config without changing pipeline call sites
- [ ] Document the round-trip from export to repository to served endpoint

## Acceptance criteria
- [ ] Generated config.pbtxt and layout validate against the documented Triton schema for a synthetic exported model
- [ ] With a mocked Triton response, the remote backend produces spans equal to the local ONNX backend on synthetic input
- [ ] Backend selection is config-driven and covered by a test

## Out of scope
- No bundling of Triton/KServe binaries (user-operated, out-of-process)
- No GPU-specific tuning
- No raw PHI sent in adapter logs

## Files
- openmed/service/backends/triton_repository.py
- openmed/service/backends/remote_inference.py
- tests/unit/service/test_triton_repository.py
- docs/serving/kserve-triton.md

---
Task: OM-472  ·  Milestone: v2.3  ·  Priority: P2  ·  Size: L
Depends on: OM-054  ·  Blocks: —
Roadmap: v2.3 rolling release train: On-device reach and ecosystem packaging
Spec: PLANS/V2/EXECUTION/tasks/OM-472.md

### PR DESCRIPTION
**Pull request by maziyarpanahi**
# Pull Request

## Description
Adds a client-only deployment and inference path for OpenMed ONNX token classifiers on user-operated Triton or KServe infrastructure. This is a fresh implementation on current `master`; it does not revive the stale, closed #1724 branch.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [x] Test addition/improvement

## Changes Made
- Generate deterministic, schema-derived `config.pbtxt` files and safe Triton repositories for single-file and external-data ONNX token classifiers, with atomic version writes, schema parity checks, and path-escape protection.
- Add KServe V2 HTTP JSON and wire-compatible gRPC transports that reshape remote logits and reuse OpenMed's local tokenizer, offset mapping, and entity decoder.
- Select the remote backend through `OpenMedConfig` without changing pipeline calls; keep it opt-in, no-egress aware, TLS-safe, and outside auto-detection.
- Add the client-only `triton` extra, generated protocol subset, public docs, NOTICE attribution, changelog entry, and synthetic mocked parity/regression coverage.

## Testing
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have tested this change with different models/inputs

Exact validation:
- `.venv/bin/python -m pytest tests/ -q` — 12,309 passed, 102 skipped
- `.venv/bin/python -m pytest tests/unit/service/test_triton_repository_adapter.py -q` — 22 passed
- `uv run --frozen make format`
- `uv run --frozen make lint`
- `uv run --frozen make format-check`
- `uv run --frozen make type-check`
- `.venv/bin/python scripts/generate_grpc_stubs.py --check`
- `.venv/bin/python scripts/release/check_public_api.py`
- `.venv/bin/python scripts/release/check_license_policy.py`
- `uv lock --check`
- `uv run --frozen make docs-build`
- `uv run --with build --frozen make build`

The acceptance tests cover documented Triton protobuf-text fields, single-file and external-data layouts, immutable/matching versions, malicious sidecar paths and symlinks, mocked HTTP/gRPC payloads, malformed logits, offline blocking, config/TOML selection, and exact span parity with the local ONNX pipeline. No live server or network is used.

## Documentation
- [x] I have updated the documentation accordingly
- [x] I have added docstrings to new functions/classes
- [x] I have updated the CHANGELOG.md

## Code Quality
- [x] I ran `make format`, `make lint`, and `make format-check`
- [ ] For Swift/OpenMedKit changes, I ran `make format-swift` and `make lint-swift`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

No Swift/OpenMedKit surface is changed.

## Dependencies
- [ ] I have not added any new dependencies
- [x] OR I have added new dependencies and they are justified because: the optional `triton` extra groups `grpcio`, `httpx`, `numpy`, `protobuf`, and `transformers`, all already used elsewhere in OpenMed, to implement client-side protocol transport and local tokenization. It deliberately includes no Triton server, KServe server, `tritonclient`, or other serving runtime. The repository license-policy gate passes.

## Checklist
- [x] I have read the contributing guidelines
- [x] My commits have clear, descriptive messages
- [x] I have squashed/organized my commits appropriately

## Related Issues
Closes #829

## Screenshots/Examples
Not applicable. The documented round trip and generated repository trees are in `docs/serving/kserve-triton.md`.

### CODE REVIEW

**Comment by maziyarpanahi** [comment]
Reviewed the exact current-master head. Local KServe/Triton and service suite: 107 passed, 5 skipped; Ruff and lock checks passed. Hosted checks: all successful.
